src/Http/Routing/src/Builder/RoutingEndpointConventionBuilderExtensions.cs

```diff
@@ -40,6 +40,57 @@ public static class RoutingEndpointConventionBuilderExtensions
             return builder;
         }
 
+        /// <summary>
+        /// Requires that endpoints match the specified header during routing.
+        /// </summary>
+        /// <param name="builder">The <see cref="IEndpointConventionBuilder"/> to add the metadata to.</param>
+        /// <param name="headerName">Header name to match.</param>
+        /// <param name="headerValues">
+        /// Acceptable header values that should match.
+        /// An empty collection means any value will be accepted as long as the header is present.
+        /// </param>
+        /// <returns>A reference to this instance after the operation has completed.</returns>
+        public static TBuilder RequireHeader<TBuilder>(this TBuilder builder, string headerName, params string[] headerValues) where TBuilder : IEndpointConventionBuilder
+        {
+            return RequireHeader(builder, headerName, HeaderValueMatchMode.Exact, false, headerValues);
```

Comment:
I think matching should be case insensitive by default to be inline with the rest of routing